sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobServiceAsyncClient.java

```diff
@@ -244,6 +230,73 @@ public BlobServiceVersion getServiceVersion() {
+     *
+     * @param containerName Name of the container to create
+     * @return A {@link Mono} containing a {@link BlobContainerAsyncClient} used to interact with the container created,
+     * or null if the container already exists.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<BlobContainerAsyncClient> createBlobContainerIfNotExists(String containerName) {
+        try {
+            return createBlobContainerIfNotExistsWithResponse(containerName, null, null).flatMap(FluxUtil::toMono);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Creates a new container within a storage account if it does not exist. For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/create-container">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <!-- src_embed com.azure.storage.blob.BlobServiceAsyncClient.createBlobContainerIfNotExistsWithResponse#String-Map-PublicAccessType -->
+     * <pre>
+     * Map&lt;String, String&gt; metadata = Collections.singletonMap&#40;&quot;metadata&quot;, &quot;value&quot;&#41;;
+     *
+     * BlobContainerAsyncClient containerClient = client
+     *     .createBlobContainerIfNotExistsWithResponse&#40;&quot;containerName&quot;, metadata, PublicAccessType.CONTAINER&#41;.block&#40;&#41;.getValue&#40;&#41;;
+     * </pre>
+     * <!-- end com.azure.storage.blob.BlobServiceAsyncClient.createBlobContainerIfNotExistsWithResponse#String-Map-PublicAccessType -->
+     *
+     * @param containerName Name of the container to create
+     * @param metadata Metadata to associate with the container. If there is leading or trailing whitespace in any
+     * metadata key or value, it must be removed or encoded.
+     * @param accessType Specifies how the data in this container is available to the public. See the
+     * x-ms-blob-public-access header in the Azure Docs for more information. Pass null for no public access.
+     * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains a {@link
+     * BlobContainerAsyncClient} used to interact with the container created, or null if the container already exists.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<BlobContainerAsyncClient>> createBlobContainerIfNotExistsWithResponse(String containerName,
+        Map<String, String> metadata, PublicAccessType accessType) {
+        try {
+            return withContext(context -> createBlobContainerIfNotExistsWithResponse(containerName, metadata, accessType,
+                context));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    Mono<Response<BlobContainerAsyncClient>> createBlobContainerIfNotExistsWithResponse(String containerName,
+        Map<String, String> metadata, PublicAccessType accessType, Context context) {
```

Comment:
This looks like it's calling into itself in an infinite loop? 

This should be creating a container client and then calling the method on that object. You can see the createBlobContainerWithResponse method in this same file for an example

Reply:
Same for delete

Reply:
I don't think it's calling itself, this method calls into createBlobContainerIfNotExistsWithResponse(containerName, metadata, accessType, context). I call withContext which calls into the next overload method.